beacon.go

```diff
@@ -30,25 +31,41 @@ import (
 )
 
 type beaconConfig struct {
-	projectID string
-	tableId   string
+	ApiVersion string
+	ProjectID  string
+	TableID    string
 }
 
 const (
-	projectKey = "GOOGLE_CLOUD_PROJECT"
-	bqTableKey = "GOOGLE_BIGQUERY_TABLE"
+	apiVersionKey = "BEACON_API_VERSION"
+	projectKey    = "GOOGLE_CLOUD_PROJECT"
+	bqTableKey    = "GOOGLE_BIGQUERY_TABLE"
 )
 
-var config = beaconConfig{
-	projectID: os.Getenv(projectKey),
-	tableId:   os.Getenv(bqTableKey),
-}
+var (
+	aboutTemplate = template.Must(template.ParseFiles("about.xml"))
+	config        = beaconConfig{
+		ApiVersion: os.Getenv(apiVersionKey),
+		ProjectID:  os.Getenv(projectKey),
+		TableID:    os.Getenv(bqTableKey),
+	}
+)
 
 func init() {
-	http.HandleFunc("/", handler)
+	http.HandleFunc("/", aboutBeacon)
+	http.HandleFunc("/query", query)
+}
+
+func aboutBeacon(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		http.Error(w, fmt.Sprintf("HTTP method %s not supported", r.Method), http.StatusMethodNotAllowed)
```

Comment:
So, 405 actually has an annoying requirement that we respond with an Allow header indicating the supported methods.  So you can keep the message but change the status code to StatusBadRequest

Reply:
done